src/scripts/deposit.js

```diff
@@ -0,0 +1,75 @@
+import * as config from '../assets/config/constants';
+
+const ethers = require('ethers');
+
+/**
+ * executeDeposit will instantiate the deposit contract & generate the deposit tx.
+ * @param {*} provider 
+ * @param {*} amount 
+ * @param {*} network 
+ * @param {*} pubKey 
+ */
+const executeDeposit = async (provider, amount, network, pubKey) => { 
+  const { contract, txCount } = await instantiateDepositContract(provider, pubKey, network);
+  const txHash = await generateDepositTx(contract, amount, txCount, pubKey, provider, network);
+  return { txHash, contract };
+};
+
+/**
+ * instantiateDepositContract returns the instantiated contract, transaction count, and signer.
+ * @param {*} provider 
+ * @param {*} pubKey 
+ * @param {*} network 
+ */
+const instantiateDepositContract = async (provider, pubKey, network) => {
+  const txCount = await provider.getTransactionCount(pubKey);
+  const signer = provider.getSigner();
+  const contractAddress = getDepositContractAddress(network);
+  if (contractAddress !== null) {
+    let contract = new ethers.Contract(contractAddress, config.DEPOSIT_CONTRACT_ABI, signer);
+    return { contract, txCount, signer };
+  }
+};
+
+/**
+ * getDepositContractAddress returns the contract address for the given chain id.
+ * @param {*} network 
+ */
+const getDepositContractAddress = (network) => {
+  let contractAddress;
+  if (network === '3') {
+    contractAddress= config.DEPOSIT_CONTRACT_ADDRESS_ROPSTEN;
+  } else if (network === '42') {
+    contractAddress = config.DEPOSIT_CONTRACT_ADDRESS_KOVAN;
+  } else if (network === '4') {
+    contractAddress = config.DEPOSIT_CONTRACT_ADDRESS_RINKEBY;
+  } else {
+    return null; //handle this error
```

Comment:
how is this handled for other networks?

Reply:
The UI will not allow you to select any network other that ropsten, rinkeby, or kovan and submit a TX so this code block is more so ensuring that the correct address is returned and if no address is present for whatever edge case, it will not instantiate the contract. I will add better error handling & display of errors. @5chdn